src/Helper/GitHelper.php

```diff
@@ -199,6 +199,31 @@ function ($commitHash) {
     }
 
     /**
+     * @param string $remote Remote name or git-repository url
+     * @param string $branch
+     *
+     * @return bool
+     */
+    public function remoteBranchExists($remote, $branch)
+    {
+        $result = $this->processHelper->runCommand(['git', 'ls-remote', $remote], true);
+
+        return 1 === preg_match('#(?<=\s)'.preg_quote('refs/heads/'.$branch, '#').'(?!\w)#', $result);
```

Comment:
I needed the look(behind,ahead) because else it would match also match a part of the branch (see description above).